moped-editor/src/views/dashboard/DashboardView.js

```diff
@@ -125,47 +211,66 @@ const DashboardView = () => {
     console.log(error);
   }
 
-  let selectedData = [];
+  const [activeTab, setActiveTab] = useState(0);
 
-  if (TABS[activeTab].label === "Following" && !!data) {
-    selectedData = data.moped_user_followed_projects;
-  } else if (TABS[activeTab].label === "My projects" && !!data) {
-    selectedData = data.moped_proj_personnel;
-  }
+  // rows and rowModesModel used in DataGrid
+  const [rows, setRows] = useState([]);
+  const [rowModesModel, setRowModesModel] = useState({});
 
-  if (selectedData) {
-    /**
-     * Build data needed in Dashboard Material Table
-     */
-    selectedData.forEach((project) => {
-      project["project_name_full"] = project.project.project_name_full;
-      project["project_id"] = project.project.project_id;
-      project["phase_name"] =
-        project.project.moped_proj_phases?.[0]?.moped_phase.phase_name;
-      project["phase_key"] =
-        project.project.moped_proj_phases?.[0]?.moped_phase.phase_key;
-      project["current_phase_id"] =
-        project.project.moped_proj_phases?.[0]?.moped_phase.phase_id;
+  // sets the data grid row data when query data is fetched
+  useEffect(() => {
+    let selectedData = [];
+
+    if (TABS[activeTab].label === "Following" && !!data) {
+      selectedData = data.moped_user_followed_projects;
+    } else if (TABS[activeTab].label === "My projects" && !!data) {
+      selectedData = data.moped_proj_personnel;
+    }
+
+    if (selectedData) {
       /**
-       * Get percentage of milestones completed
+       * Build data needed in Dashboard Material Table
```

Comment:
Is there a way to memoize this, or just build once and not every time a tab is clicked? When I use prod data locally, moped thinks I'm Ivonne and as a result I have a ton of records in the My Projects tab. Clicking back and forth between the tabs, it feels a tiny bit sluggish.

Reply:
this is definitely doable and worth it in my opinion! i'll think about how to do this as simply as possible, although if it's more complex it might be best to spin it off into a separate issue as an enhancement for the subsequent release

Reply:
I was thinking about this, and how its not a regression since its currently behaving like this in prod. this is an improvement because of the useEffect, so spinoff makes sense so we arent rushed